mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -28,17 +28,21 @@
 
     private Configuration configuration;
 
+    private MavenProjectConfigurator mavenProjectConfigurator;
```

Comment:
It is slightly confusing - storing the `MavenProjectConfigurator` as a global variable just because of reading system properties. If you retrieve the information from system properties, then it can be used as a `public static` method from separated class. The other case would be storing a boolean flag as a information that the ST tool has been initialized (based on information that it has (not) been disabled or the test has (not) been skipped)